### PR TITLE
Refactor to move some email/sms code to the AccountWorkflowService

### PR DESCRIPTION
**Pull request by alxdarksage**
This code is more closely related to the workflow service, and in order to provide sign in links to the account exists email, without creating a circular dependency with AuthenticationService, I'm refactoring to clean this up first.

### CODE REVIEW

File: app/org/sagebionetworks/bridge/services/AccountWorkflowService.java [hunk 1]
```diff
@@ -265,7 +290,103 @@ public void resetPassword(PasswordReset passwordReset) {
...
+            token = tokenSupplier.get();
+            cacheProvider.setObject(cacheKey, token, SESSION_SIGNIN_EXPIRE_IN_SECONDS);
+        }
+
+        messageSender.accept(study, token);
+        atomicLong.set(System.currentTimeMillis()-startTime);
+    }
 
+    public AccountId channelSignIn(ChannelType channelType, CriteriaContext context, SignIn signIn,
+            Validator validator) {
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/AccountWorkflowService.java]: Are there unit tests for channelSignIn?

File: test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java [hunk 2]
```diff
@@ -304,130 +245,59 @@ public void signOutNoSessionToken() {
...
+
         UserSession retSession = service.emailSignIn(CONTEXT, SIGN_IN_WITH_EMAIL);
         
         assertNotNull(retSession);
         assertEquals(REAUTH_TOKEN, retSession.getReauthToken());
-        verify(accountDao, never()).changePassword(eq(account), any());
         verify(accountDao).getAccountAfterAuthentication(SIGN_IN_WITH_EMAIL.getAccountId());
         verify(accountDao).verifyChannel(AuthenticationService.ChannelType.EMAIL, account);
-        verify(cacheProvider).removeObject(CACHE_KEY);
-        verify(cacheProvider).setUserSession(retSession);
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java]: cacheProvider.setUserSession() is still being called.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: app/org/sagebionetworks/bridge/services/AccountWorkflowService.java (hunk 1 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/services/AccountWorkflowService.java]: Yes I noticed these were missing and wrote them yesterday. Added to this PR.

File: test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java (hunk 2 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java]: Added this back, verified that session is set for ConsentRequiredException. Also added a test of consent setting a session for a phone sign in.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**